public-charge/scraper/scraper.py

```diff
@@ -0,0 +1,178 @@
+        return results_available
+
+    def get_comments_urls_on_page(self, page_number):
+        """ Get a list of all comments on a given page
+
+        INPUTS
+        ------
+        page_number: int
+
+        RETURNS
+        -------
+        comments: list of strings
+            Each entry in the list corresponds to the url for a comment.
+        """
+        # Get url for given page.
+        url = Scraper.get_page(page_number=page_number)
+
+        # Open browser.
+        self.driver.get(url)
+        sleep(self.delay)
+
+        # Get comment urls.
+        elems = self.driver.find_elements_by_xpath("//a[@href]")
+        comments = [
+            e.get_attribute("href")
+            for e in elems
+            if "document?D=USCIS-" in e.get_attribute("href")
+        ]
+
+        return list(set(comments))
+
+    def scrape_comment(self, comment_url):
+        """
+        INPUTS
+        ------
+        comment_url: str
+            URL for a comment.
+
+        RETURNS
+        -------
+        results: dict
+            Name, comment, posted and received date, tracking number, and rin.
+        """
+        # Go to comment.
+        self.driver.get(comment_url)
+        sleep(self.delay)
+
+        # Get name.
+        name = self.driver.find_element_by_xpath(
+            '//*[contains(text(), "Comment Submitted by")]').text
+
+        # Check that text is not empty.
+        try:
+            comment_text = self.driver.find_element_by_xpath(
+                '//div[contains(text(), "View document")]/../following-sibling::div'
+            ).text
+        except:
+            comment_text = self.driver.find_element_by_xpath(
+                '//div[@class="GIY1LSJIXD"]').text
+
+        # Get date posted.
+        posted_date = self.driver.find_element_by_xpath(
+            '//span[contains(@title, "Date the document is posted")]/..').text
+        posted_date = posted_date.split(':\n')[-1]
+
+        # Get tracking number.
+        tracking_number = self.driver.find_element_by_xpath(
+            '//*[contains(text(), "Tracking Number")]/../following-sibling::span'
+        ).text
+
+        # Get RIN.
+        rin = self.driver.find_element_by_xpath(
+            '//*[contains(text(), "RIN")]/../following-sibling::span').text
+
+        # Get received date.
+        self.driver.find_element_by_xpath(
+            '//a[contains(text(), "Show More Details")]').click()
+        received_date = self.driver.find_element_by_xpath(
+            '//span[@title="The date the agency received or created the document"]/..'
+        ).text
+        received_date = received_date.split(':')[-1]
+
+        results = {
+            "name": name,
+            "comment": comment_text,
+            "posted_date": posted_date,
+            "received_date": received_date,
+            "tracking_number": tracking_number,
+            "rin": rin,
+        }
+        return results
+
+
+if __name__ == "__main__":
```

Comment:
Does the Scrapper need this main method if the class is used in the `get_comments.py` script we intend to run? If it doesn't, I'd recommend removing it for clarity.

Reply:
Got it, thanks!